whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -205,10 +227,44 @@ private Boolean isRedirect(ObjectType objectType, final String key) {
     }
 
     protected Response lookupObject(final HttpServletRequest request, final Set<ObjectType> objectTypes, final String key) {
-        List<RpslObject> result =  rdapQueryHandler.handleQuery(getQueryObject(objectTypes, key), request);
+        final Iterable<RpslObject> result =  rdapQueryHandler.handleQuery(getQueryObject(objectTypes, key), request);
         return getResponse(request, result);
     }
 
+    protected Response lookupForOrganisation(final HttpServletRequest request, final Set<ObjectType> objectTypes,
+                                             final String key) {
+        final List<RpslObject> organisationResult = rdapQueryHandler.handleQuery(getQueryObject(objectTypes,
+                        key),
+                request);
+
+        if (organisationResult.isEmpty()){
+            throw new NotFoundException("Requested organisation not found: " + key);
+        }
+        if (organisationResult.size() > 1){
+            throw new IllegalStateException("Unexpected result size: " + organisationResult.size());
+        }
+
+        final Query autnumInetnumForOrganisationQuery = Query.parse(
+                String.format("%s %s %s %s %s %s %s %s",
+                        QueryFlag.NO_GROUPING.getLongFlag(),
+                        QueryFlag.NO_REFERENCED.getLongFlag(),
+                        QueryFlag.SELECT_TYPES.getLongFlag(),
+                        objectTypesToString(List.of(AUT_NUM, ObjectType.INETNUM, ObjectType.INET6NUM)),
+                        QueryFlag.NO_FILTERING.getLongFlag(),
+                        QueryFlag.INVERSE.getLongFlag(),
+                        AttributeType.ORG.getName(),
+                        key));
+
+        final Stream<RpslObject> resourcesResult = rdapQueryHandler.handleQueryStream(autnumInetnumForOrganisationQuery,
+                request);
+        return getOrganisationResponse(request, joinResults(organisationResult, resourcesResult));
+    }
```

Comment:
What is the use of joining the results? You can get rid of switch statements in getOrganisationResponse if you pass it separately.  We are searchimng for organisation objects and later on resources underneath it, we should not join them in one list and later on seperate them